### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by mattathompson**
# Work in progress

- [x] Install & Setup initial typescript configuration.
- [x] Move over `.js` files to `.ts` files as is w/ passing specs and minimal changes.
- [x] Update `bin/solidarity` to point at the new dist directory. 

### Local setup should just include `npm install -g typescript`

### CODE REVIEW

**Comment by GantMan** [comment]
Code coverage drop is fine.  There will be a sedulous effort to increase coverage near release of 1.0

**Comment by GantMan** [comment]
This is looking really good btw!  Sorry code is changing while you're working.

**Comment by skellock** [comment]
Would y'all be open to adding `dist` to `.gitignore` and then using `files` in `packages.json` for the npm manifest?  😻

**Comment by GantMan** [comment]
I agree. `dist` in `.gitignore`, but you can just add `src` to `.npmignore` rather than having to use files.

**Comment by skellock** [comment]
I prefer opt-in rather than opt-out.  My $0.02.  =)

**Comment by mattathompson** [comment]
@skellock, @GantMan I'll look into implementing the suggestions this evening if I have time. Once thats done and I resolve the conflicts this should be good to review.

**Comment by mattathompson** [comment]
@GantMan We'll need to add these as a setup steps for the **semaphoreci**:

1. `npm install -g typescript` 
2. `tsc` (to build the dist)

**Comment by GantMan** [comment]
@mattathompson - sounds good!
Are you done?  I see there are conflicts (this project is just growing so fast!).  If you'd like I can help resolve them, but I want to make sure you're ready before I dive in.

It's probably just the past few merged PRs anyway.

**Comment by GantMan** [comment]
Also, this is an epic PR... I'm going to talk to our Ops and see if they can send you an infinite red mug.

**Comment by mattathompson** [comment]
Yeah, I merged master last night. I'm on the train now but I'll
merge/resolve when I get into the office.

**Comment by mattathompson** [comment]
Haha that would be sick!

**Comment by GantMan** [comment]
When you get a chance, join our public slack http://community.infinite.red and shoot me a message with your details, and I'll get the very slow but promising IR mug ball rolling.

**Comment by mattathompson** [comment]
@GantMan Merged/Resolved

File: src/commands/snapshot.js [hunk 1]
```diff
@@ -1,93 +1,96 @@
...
-    filesystem.copy(
-      `${runPlugin.templateDirectory}${runPlugin.snapshot}`,
-      '.solidarity'
-    )
-    // force local version update
-    await system.run('solidarity snapshot')
-  } else {
-    // run plugin's snapshot function
-    await runPlugin.snapshot(context)
+namespace Snapshot {
```
Comment (GantMan) [inline src/commands/snapshot.js]: you keeping this js file?

Comment (mattathompson) [reply src/commands/snapshot.js]: Whoops. Happened in the merge I guess.

**Pull request merged**